mc/watch.go

```diff
@@ -119,14 +115,24 @@ func (w *Watcher) Wait() {
 	w.wg.Wait()
 }
 
-// Join the watcher with client
-func (w *Watcher) Join(client Client, recursive bool) *probe.Error {
```

Comment:
We will not have such clients. We shouldnt bring in new interfaces in such a manner.